### PR TITLE
Made generateEmptyCalendar and fillCalendar into Checkit methods

### PR DESCRIPTION
**Pull request by andrayantelo**
This branch I created from master, and then I merged the checkmarkClickHandler branch into it, which is the branch where I worked on making the checkmark click handler something that the checkit object handles. 

So, now I can generate an empty calendar, fill it, and attach the click handler in the checkit object. Am missing the function that generates in checkmarks (like when you have saved checkmarks), and the function that removes empty weeks. Still need to make those two into checkit object methods. 

This PR is already huge.. so, how should I proceed? Wait til you approve this one? Does the other PR even matter anymore (The one where I made the click Handler into something that is handled by the checkit object). @johntyree

### CODE REVIEW

File: docs/checkit.js [hunk 1]
```diff
@@ -354,12 +356,92 @@ CheckIt.prototype.onAuthStateChanged = function(user) {
...
-CheckIt.prototype.checkSignedInWithMessage = function() {
-    // Return true if the user is signed in Firebase
-    if (this.auth.currentUser) {
-        return true;
-    }
+CheckIt.prototype.generateEmptyCalendar = function(calObj) {
+    // Generate the html for an empty calendar of the calendar you want to 
+    // display.
+    
+    var calendarDiv = this.$calendarDiv;
```
Comment (johntyree) [inline docs/checkit.js]: The point of prefixing variables with `$` is so you know whether or not it's already jquery selector.

File: docs/checkit.js [hunk 2]
```diff
@@ -354,12 +356,92 @@ CheckIt.prototype.onAuthStateChanged = function(user) {
...
+    
+    // Add the title of the calendar
+    calendarDiv.append('<div id="calendarTitleHeading"> <h1 class="page-header text-center">' +
+                        calObj.state.title + '</h1></div>');
+                        
+
+    calObj.monthObjects.forEach (function(monthObj, index) {
+
+        var yearHeader = "<div class='page-header text-center'>" +
+                         "<h2 id='yearHeader'>" + monthObj.monthYear + "</h2>" +
```
Comment (johntyree) [inline docs/checkit.js]: My understanding is that `id` attributes should be unique across the page.

File: docs/checkit.js [hunk 3]
```diff
@@ -354,12 +356,92 @@ CheckIt.prototype.onAuthStateChanged = function(user) {
...
+                         "</div>";
+        
+        //the div ID is the monthID
+            
+        calendarDiv.append('<div class="monthframe" id=' + monthObj.monthId + '></div>');
+        
+        if (self.monthIndex === 0) {
+            $('#' + monthObj.monthId).append(yearHeader);
```
Comment (johntyree) [inline docs/checkit.js]: Why create the year header outside of the this conditional?

File: docs/checkit.js [hunk 4]
```diff
@@ -354,12 +356,92 @@ CheckIt.prototype.onAuthStateChanged = function(user) {
...
+
+CheckIt.prototype.fillCalendar = function(calObj) {
+    // Fill an empty calendar with appropriate calendar data.
+
+    calObj.monthObjects.forEach (function(monthObj) {
+        
+        var $monthId = $('#'+ monthObj.monthId);
+        
+        $monthId.find(".month-year").empty();
+        $monthId.find(".month-year").append(monthObj.monthName + " " + monthObj.monthYear);
```
Comment (johntyree) [inline docs/checkit.js]: Are these two lines together are the same as `$monthId.find(".month-year").text(...)`?

File: docs/checkit.js [hunk 5]
```diff
@@ -354,12 +356,92 @@ CheckIt.prototype.onAuthStateChanged = function(user) {
...
+        
+        // Go through each td and fill in correct day number
```
Comment (johntyree) [inline docs/checkit.js]: What is a td here?

File: docs/checkit.js [hunk 6]
```diff
@@ -576,10 +658,11 @@ CheckIt.prototype.buildCalendar = function(calendarObject) {
     //this function assumes the calendarObject already has it's
     //state updated with the correct information. 
     
-    calendarObject.generateEmptyCalendar(calendarObject.monthObjects);
-    calendarObject.fillCalendar(calendarObject.monthObjects);
-    this.attachCellClickHandler(calendarObject, calendarObject.monthObjects);
-    calendarObject.generateCheckmarks();
+    this.generateEmptyCalendar(calendarObject);
+    this.fillCalendar(calendarObject);
+    //calendarObject.fillCalendar(calendarObject.monthObjects);
```
Comment (johntyree) [inline docs/checkit.js]: remove commented code

**Review by johntyree — CHANGES_REQUESTED** [review]

File: docs/checkit.js (hunk 2 above)
Comment (andrayantelo) [reply docs/checkit.js]: Don't actually need that id, since bootstrap classes take care of the CSS. Removed it.

File: docs/checkit.js (hunk 3 above)
Comment (andrayantelo) [reply docs/checkit.js]: No reason. Moved it.

File: docs/checkit.js (hunk 5 above)
Comment (andrayantelo) [reply docs/checkit.js]: the element <td></td> which has the .cell div in it which contains the div that has the day number in it and the checkmark div.

**Review by johntyree — APPROVED** [review]

**Review by johntyree — APPROVED** [review]

**Pull request merged**